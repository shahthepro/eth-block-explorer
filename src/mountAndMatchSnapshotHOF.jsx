export default function (it, ReactDOM, mount) {
  return function (elementName, element) {
    it(`renders <${elementName}> without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(element, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
    it(`matches <${elementName}> snapshot`, () => {
      const mountedElement = mount(element);
      expect(mountedElement).toMatchSnapshot();
    });
  }
}