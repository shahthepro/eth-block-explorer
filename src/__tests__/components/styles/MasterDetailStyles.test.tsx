import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {
  MasterContainer,
  DetailsContainer,
  MasterDetailWrapper,
} from './../../../components/styles/MasterDetailStyles';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import mountAndMatchSnapshotHOF from '../../../../mountAndMatchSnapshotHOF';

Enzyme.configure({ adapter: new Adapter() })
const { mount } = Enzyme;

const mountAndMatchSnapshot = mountAndMatchSnapshotHOF(it, ReactDOM, mount);

mountAndMatchSnapshot('MasterContainer', <MasterContainer />);
mountAndMatchSnapshot('DetailsContainer', <DetailsContainer />);
mountAndMatchSnapshot('MasterDetailWrapper', <MasterDetailWrapper />);