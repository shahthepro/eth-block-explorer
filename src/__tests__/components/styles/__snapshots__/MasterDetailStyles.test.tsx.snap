// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches <DetailsContainer> snapshot 1`] = `
<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "cXclpg",
          "rules": Array [
            "
        flex: auto;
        transition: flex 0.3s ease;
      ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-htpNat cXclpg"
    />
  </StyledComponent>
</styled.div>
`;

exports[`matches <MasterContainer> snapshot 1`] = `
<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "lastClassName": "lacFfN",
          "rules": Array [
            "
        flex: 280px 0 0;
        transition: flex 0.3s ease, opacity 0.3s ease 0.3s;
        opacity: 1;
        &.master-closed {
          flex: 0px 0 0;
          opacity: 0;
        }
      ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-bwzfXH lacFfN"
    />
  </StyledComponent>
</styled.div>
`;

exports[`matches <MasterDetailWrapper> snapshot 1`] = `
<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "kxKaOB",
          "rules": Array [
            "
        height: 100%;
        width: 100%;
        display: flex;
        flex-direction: row;
        background-color: #fff;
        &.limit-height {
          max-height: 580px;
        }
        &.auto-height {
          height: auto;
        }
      ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-bdVaJa kxKaOB"
    />
  </StyledComponent>
</styled.div>
`;
